legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-bigquery/src/main/java/org/finos/legend/engine/persistence/components/relational/bigquery/BigQuerySink.java

```diff
@@ -0,0 +1,226 @@
+import java.util.Optional;
+import java.util.Set;
+
+public class BigQuerySink extends AnsiSqlSink
+{
+    private static final RelationalSink INSTANCE;
+    private static final Set<Capability> CAPABILITIES;
+    private static final Map<Class<?>, LogicalPlanVisitor<?>> LOGICAL_PLAN_VISITOR_BY_CLASS;
+    private static final Map<DataType, Set<DataType>> IMPLICIT_DATA_TYPE_MAPPING;
+    private static final Map<DataType, Set<DataType>> EXPLICIT_DATA_TYPE_MAPPING;
+
+    static
+    {
+        Set<Capability> capabilities = new HashSet<>();
+        capabilities.add(Capability.MERGE);
+        capabilities.add(Capability.ADD_COLUMN);
+        capabilities.add(Capability.IMPLICIT_DATA_TYPE_CONVERSION);
+        capabilities.add(Capability.EXPLICIT_DATA_TYPE_CONVERSION);
+        capabilities.add(Capability.DATA_TYPE_LENGTH_CHANGE);
+        capabilities.add(Capability.DATA_TYPE_SCALE_CHANGE);
+        CAPABILITIES = Collections.unmodifiableSet(capabilities);
+
+        Map<Class<?>, LogicalPlanVisitor<?>> logicalPlanVisitorByClass = new HashMap<>();
+        logicalPlanVisitorByClass.put(SchemaDefinition.class, new SchemaDefinitionVisitor());
+        logicalPlanVisitorByClass.put(Create.class, new SQLCreateVisitor());
+        logicalPlanVisitorByClass.put(ClusterKey.class, new ClusterKeyVisitor());
+        logicalPlanVisitorByClass.put(PartitionKey.class, new PartitionKeyVisitor());
+        logicalPlanVisitorByClass.put(Alter.class, new AlterVisitor());
+        logicalPlanVisitorByClass.put(Delete.class, new DeleteVisitor());
+        logicalPlanVisitorByClass.put(Field.class, new FieldVisitor());
+        logicalPlanVisitorByClass.put(Truncate.class, new TruncateVisitor());
+        logicalPlanVisitorByClass.put(BatchEndTimestamp.class, new BatchEndTimestampVisitor());
+        logicalPlanVisitorByClass.put(BatchStartTimestamp.class, new BatchStartTimestampVisitor());
+        LOGICAL_PLAN_VISITOR_BY_CLASS = Collections.unmodifiableMap(logicalPlanVisitorByClass);
+
+        Map<DataType, Set<DataType>> implicitDataTypeMapping = new HashMap<>();
+        implicitDataTypeMapping.put(DataType.INTEGER, new HashSet<>(Arrays.asList(DataType.INT, DataType.BIGINT, DataType.TINYINT, DataType.SMALLINT, DataType.INT64)));
+        implicitDataTypeMapping.put(DataType.NUMERIC, new HashSet<>(Arrays.asList(DataType.NUMERIC, DataType.NUMBER, DataType.DECIMAL, DataType.INT, DataType.INTEGER, DataType.BIGINT, DataType.TINYINT, DataType.SMALLINT, DataType.INT64)));
+        implicitDataTypeMapping.put(DataType.FLOAT, new HashSet<>(Arrays.asList(DataType.REAL, DataType.DOUBLE, DataType.FLOAT64, DataType.INT, DataType.INTEGER, DataType.BIGINT, DataType.TINYINT, DataType.SMALLINT, DataType.INT64, DataType.NUMBER, DataType.NUMERIC, DataType.DECIMAL)));
+        implicitDataTypeMapping.put(DataType.STRING, new HashSet<>(Arrays.asList(DataType.STRING, DataType.CHAR, DataType.CHARACTER, DataType.VARCHAR, DataType.LONGNVARCHAR, DataType.LONGTEXT, DataType.TEXT)));
```

Comment:
These HashSet should be wrappd on unmodifiable to actually make the IMPLICIT_DATA_TYPE_MAPPING fully unmodifiable.

Reply:
Thanks, Changed as suggested